packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_list.dart

```diff
@@ -114,53 +114,47 @@ class _SnapshotListItems extends StatefulWidget {
 class _SnapshotListItemsState extends State<_SnapshotListItems>
     with AutoDisposeMixin {
   final _headerHeight = 1.20 * defaultRowHeight;
-  late ScrollController _scrollController;
-  late int _currentSnapshotsLength;
+  late final ScrollController _scrollController;
 
   @override
   void initState() {
-    super.initState();
     _scrollController = ScrollController();
+    super.initState();
   }
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
 
-    _currentSnapshotsLength = widget.controller.snapshots.value.length;
-
     addAutoDisposeListener(widget.controller.snapshots);
 
-    addAutoDisposeListener(widget.controller.selectedIndex, () async {
-      setState(() {});
+    addAutoDisposeListener(widget.controller.selectedIndex);
 
-      final newSnapshotsLength = widget.controller.snapshots.value.length;
-      if (newSnapshotsLength > _currentSnapshotsLength) {
-        _currentSnapshotsLength = newSnapshotsLength;
-        await _scrollController.autoScrollToBottom();
-        setState(() {});
-      }
+    addAutoDisposeListener(widget.controller.itemAdded, () async {
+      await _scrollController.autoScrollToBottom();
```

Comment:
we don't need another notifier for this. We are already notifying when an item is added because the widget.controller.snapshots notifier will be triggered.